### DISCUSSION TITLE
Custom Items

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#908] [Help] [answered]
Hello, is it possible to create your own custom ItemStack class?
![image](https://user-images.githubusercontent.com/71212639/162811339-e0f1d207-cddd-436e-a0ee-4592717cbcdc.png)
something like this, but than something that will work.
~Jochem

**Comment by TheMode** [comment, marked as answer]
This is not possible due to how the protocol is designed. Minestom must be able to re-create items from its material, amount & nbt. 
Custom `ItemStack` implementations have the risk to be discarded at any time.

In any case, there should not be any reason to make your own implementation, take a look at the [Tag API](https://wiki.minestom.net/feature/tags) for applying data

**Reply by DevBlokJochem** [reply]
I know that that api exists, but a custom itemstack makes it mutch easier.

**Reply by TheMode** [reply]
It would also make it very difficult for third party (extensions, …) to use your items. And prevent us from guaranteeing immutability.

**Reply by DevBlokJochem** [reply]
ofc, but thats the reason why Im using this only for my own core.  A quick other question. How can I add a tagg? (I can't find that part on the tutorial.)

**Reply by TheMode** [reply]
`net.minestom.server.tag`

**Reply by sqcred** [reply]
> ofc, but thats the reason why Im using this only for my own core. A quick other question. How can I add a tagg? (I can't find that part on the tutorial.)

You can make a new itemstack and call .withTag
